qiskit/circuit/quantumcircuit.py

```diff
@@ -803,7 +804,17 @@ def append(self, instruction, qargs=None, cargs=None):
             CircuitError: if object passed is a subclass of Instruction
             CircuitError: if object passed is neither subclass nor an instance of Instruction
         """
-        # Convert input to instruction
+        # make copy of parameterized gate instance if used in another circuit
+        is_gate = isinstance(instruction, Gate)
```

Comment:
The same problem will occur for `Instruction` instances so I think the `is_gate` check is not needed.

Reply:
Agreed, I will go ahead and remove that check.